it/src/test/java/com/google/cloud/teleport/it/spanner/SpannerResourceManagerUtilsTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.spanner;
+
+import static com.google.cloud.teleport.it.spanner.SpannerResourceManagerUtils.generateDatabaseId;
+import static com.google.cloud.teleport.it.spanner.SpannerResourceManagerUtils.generateInstanceId;
+import static com.google.cloud.teleport.it.spanner.SpannerResourceManagerUtils.generateNewId;
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link com.google.cloud.teleport.it.spanner.SpannerResourceManagerUtils}. */
+@RunWith(JUnit4.class)
+public class SpannerResourceManagerUtilsTest {
+
+  @Test
+  public void testGenerateNewIdShortString() {
+    String shortId = "test-id";
+
+    String actual = generateNewId(shortId, shortId.length());
+
+    assertThat(actual).isEqualTo(shortId);
+  }
+
+  @Test
+  public void testGenerateNewIdLongString() {
+    String longId = "long-test-id-string";
+
+    String actual = generateNewId(longId, 13);
+
+    assertThat(actual).isNotEqualTo(longId);
```

Comment:
I think this is not an important check. Do we care if the generate id is equal to the base id or not? No, we don't, as long as it has the correct length and it's a valid Spanner ID.
I'd rather check that the length of the id is <= 13.